nilearn/surface/surface.py

```diff
@@ -1753,3 +1779,167 @@ def from_volume(
         data = PolyData(left=texture_left, right=texture_right)
 
         return cls(mesh=mesh, data=data)
+
+
+def check_surface_data_ndims(img, dim, var_name="img"):
+    """Check if the data of a SurfaceImage is of a given dimension,
+    raise error if not.
+
+    Parameters
+    ----------
+    img : :obj:`~nilearn.surface.SurfaceImage`
+        SurfaceImage to check.
+
+    dim : int
+        Dimensions the data should have.
+
+    var_name : str, optional
+        Name of the variable to include in the error message.
+
+    Returns
+    -------
+    raise ValueError if the data of the SurfaceImage is not of the given
+    dimension.
+    """
+    n_dim_left = img.data.parts["left"].ndim
+    n_dim_right = img.data.parts["right"].ndim
+    if not all(x == dim for x in [n_dim_left, n_dim_right]):
+        raise ValueError(
+            f"Data for each hemisphere of {var_name} should be {dim}D, "
+            f"but found {n_dim_left}D for left and {n_dim_right}D for right."
+        )
+
+
+def get_min_max_surface_image(img):
+    """Get min and max across hemisphere for a SurfaceImage.
+
+    Parameters
+    ----------
+    img : SurfaceImage
+
+    Returns
+    -------
+    vmin : float
+
+    vmax : float
+    """
+    vmin = min(min(x.ravel()) for x in img.data.parts.values())
+    vmax = max(max(x.ravel()) for x in img.data.parts.values())
+    return vmin, vmax
+
+
+def concat_extract_surface_data_parts(img):
+    """Concatenate the data of a SurfaceImage across hemispheres and return
+    as a numpy array.
+
+    Parameters
+    ----------
+    img : :obj:`~nilearn.surface.SurfaceImage` object
+        SurfaceImage whose data to concatenate and extract.
+
+    Returns
+    -------
+    :obj:`~numpy.ndarray`
+        Concatenated data across hemispheres.
+    """
+    return np.concatenate(list(img.data.parts.values()), axis=0)
+
+
+def concat_imgs(imgs):
```

Comment:
you deliberately want to have names more similar btw the surface module and the image module I guess. I find it reasonable, but the names of the different routines are less homogeneous, some with _surface_, some without.

Reply:
> but the names of the different routines are less homogeneous

can you point to an example? I am unsure what you mean by routines.

Reply:
Sorry, I mean the names of the functions in this module, that seem to be mostly ancillary, hence the name "routines": for instance, we have `mean_img` but `_extract_surface_image_data`, meaning that surface data are called surface_image or image, depending on the function.

Reply:
Any opinion on this ?

Reply:
na at the moment I would keep the name similar to those in nilearn.image when the function does the same thing.

In general I would also just call avoid using surface_image for variable names or in function since we are in the surface module.